intermediate_source/torch_export_tutorial.py

```diff
@@ -0,0 +1,462 @@
+    tb.print_exc()
+
+######################################################################
+# - unsupported Python language features (e.g. throwing exceptions, match statements)
+
+def bad4(x):
+    try:
+        x = x + 1
+        raise RuntimeError("bad")
+    except:
+        x = x + 2
+    return x
+
+try:
+    export(bad4, (torch.randn(3, 3),))
+except Exception:
+    tb.print_exc()
+
+######################################################################
+# Control Flow Ops
+# ----------------
+# ``torch.export`` actually does support data-dependent control flow.
+# But these need to be expressed using control flow ops. For example,
+# we can fix the control flow example above using the ``cond`` op, like so:
+
+from functorch.experimental import control_flow
+
+def bad1_fixed(x):
+    def true_fn(x):
+        return torch.sin(x)
+    def false_fn(x):
+        return torch.cos(x)
+    return control_flow.cond(x.sum() > 0, true_fn, false_fn, [x])
+
+exported_bad1_fixed = export(bad1_fixed, (torch.randn(3, 3),))
+print(exported_bad1_fixed(torch.ones(3, 3)))
+print(exported_bad1_fixed(-torch.ones(3, 3)))
+
+######################################################################
+# There are some limitations one should be aware of:
+# - The predicate (i.e. ``x.sum() > 0``) must result in a boolean or a single-element tensor.
+# - The operands (i.e. ``[x]``) must be tensors.
+# - The branch function (i.e. ``true_fn`` and ``false_fn``) signature must match with the
+# operands and they must both return a single tensor with the same metadata (e.g. dtype, shape, etc.)
+# - Branch functions cannot mutate inputs or globals
+# - Branch functions cannot access closure variables, except for ``self`` if the function is
+# defined in the scope of a method.
+
+######################################################################
+# We can also use ``map``, which applies a function across the first dimension
+# of the first tensor argument.
+
+from functorch.experimental.control_flow import map
+
+def map_example(xs):
+    def map_fn(x, const):
+        def true_fn(x):
+            return x + const
+        def false_fn(x):
+            return x - const
+        return control_flow.cond(x.sum() > 0, true_fn, false_fn, [x])
+    return control_flow.map(map_fn, xs, torch.tensor([2.0]))
+
+exported_map_example= export(map_example, (torch.randn(4, 3),))
+inp = torch.cat((torch.ones(2, 3), -torch.ones(2, 3)))
+print(exported_map_example(inp))
+
+######################################################################
+# Constraints
+# -----------
+# Ops can have different specializations for different tensor shapes, so
+# ``ExportedProgram``s uses constraints on tensor shapes in order to ensure
+# correctness with other inputs.
+# If we try to run the first ``ExportedProgram`` example with a tensor
+# with a different shape, we get an error:
+
+try:
+    exported_mod(torch.randn(10, 100), torch.randn(10, 100))
+except Exception:
+    tb.print_exc()
```

Comment:
Can you include a snippet of the error message as a comment?
That can help with grep/google search

Reply:
The error message will show up in the actual tutorial page, so errors can be searched for when viewing the page.